mypyc/codegen/emitfunc.py

```diff
@@ -123,6 +124,28 @@ def generate_native_function(
     for i, block in enumerate(blocks):
         block.label = i
 
+    # Find blocks that are never jumped to or are only jumped to from the
+    # block directly above it. This allows for more labels and gotos to be
+    # eliminated during code generation.
```

Comment:
Only labels are eliminated here, no?

Reply:
This code only marks the basic blocks as "referenced" if they meet the conditions above, nothing actually gets eliminated here. And yes, during codegen, labels won't be emitted for these "unreferenced" blocks, same for `goto`s (assuming they are jumping to the next block).